Add batch_get_image support for ECR

mbjerkness opened the pull request.

JackDanger commented on moto/ecr/models.py:
```diff
@@ -245,6 +256,38 @@ def put_image(self, repository_name, image_manifest, image_tag):
+                        found = True
+                        response.add(image)
+                if not found:
+                    image_id_representation = "{imageDigest:'%s', imageTag:'%s'}" % (
```
What do you think about using `json.dumps` here like in ecr/responses.py for crafting the JSON string? It might be easier to follow for future developers (I didn't immediately realize this was JSON at first)

mbjerkness commented:
@JackDanger I updated the code per your comment. You comment actually highlighted a larger issue that I added some tests for. The code wasn't responding with the right errors in some cases. It should be now.

JackDanger commented:
Nice! Thanks, @mbjerkness
